hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java

```diff
@@ -85,7 +85,7 @@ public abstract class TestOzoneManagerHA {
   /* Reduce max number of retries to speed up unit test. */
   private static final int OZONE_CLIENT_FAILOVER_MAX_ATTEMPTS = 5;
   private static final int IPC_CLIENT_CONNECT_MAX_RETRIES = 4;
-  private static final long SNAPSHOT_THRESHOLD = 50;
+  private static final long SNAPSHOT_THRESHOLD = 1;
```

Comment:
Setting this globally may mask other problems in prepare or other OM HA tests when snapshots are *not* taken. We should probably only set in the one new test.

Reply:
Due to the presence of some functions like submitCancelPrepareRequest() in setup(), the transaction index increases due to which we may not be able to produce the current scenario i.e, SNAPSHOT_THRESHOLD = 1.
Do you feel we should write a new test class extending TestOzoneManagerHA?

Reply:
I think desired result is that only the new snapshot test in `TestOzoneManagerPrepare` has a snapshot threshold of 1, and the rest of the tests for `TestOzoneManagerPrepare` and  `TestOzoneManagerHA` subclasses remain unchanged. Doing that with the current `TestOzoneManagerHA` setup is difficult because it is using a static method to set the configuration, so some refactoring may be required there.

Reply:
If we do so, the transaction index will still increase due to the setup() function in TestOzoneManagerPrepare!

Reply:
> the transaction index increases due to which we may not be able to produce the current scenario i.e, SNAPSHOT_THRESHOLD = 1.

> If we do so, the transaction index will still increase due to the setup() function in TestOzoneManagerPrepare!

I think there may be some misunderstanding about the snapshot threshold? SNAPSHOT_THRESHOLD=1 means the OM will take a snapshot on every new request, not that it will only take a snapshot on index 1. The transaction index can be any number at any time during the test, and by setting this config to 1 we know that transaction will have a snapshot taken.